File: leetcode3/황은지/3314. Construct the Minimum Bitwise Array I.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var minBitwiseArray = function(nums) {
    const arr=Array(1000).fill(-1);
    const max=Math.max(...nums);

    for(let num=0;num<=max;num++){
        const afterOr=num|(num+1);
        if(arr[afterOr]===-1) arr[afterOr]=num;
    }

    const result=Array(nums.length);
    for(let i=0;i<nums.length;i++){
        result[i]=arr[nums[i]]
    }
    
    return result;
    
};

File: leetcode3/염혜정/3314. Construct the Minimum Bitwise Array I.java
// O(n)

class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int[] result = new int[nums.size()];
        for (int i = 0; i<nums.size(); i++) {
            int num = nums.get(i);
            if (num == 2) result[i] = -1;
            else {
                for (int k = 1; k<32; k++) {
                    if (((num >> k) & 1) == 0) {
                        result[i] = num ^ (1 << (k-1));
                        break;
                    }
                }
            }
        }
        return result;
    }
}
